### PR TITLE
fix(web): upload and link asset

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
In this PR I implemented upload and link asset in schema and content page

### CODE REVIEW

File: web/src/components/molecules/Common/Form/AssetItem/hooks.ts [hunk 1]
```diff
@@ -30,10 +32,10 @@ export default (
 
   const handleUpload = useCallback(async () => {
     setUploading(true);
-    await createAssets(fileList);
+    const assets = await createAssets(fileList);
+    if (assets?.length > 0) onLink(assets[0]);
```
Comment (KaWaite) [inline web/src/components/molecules/Common/Form/AssetItem/hooks.ts:36]: Shouldn't we limit to only a single asset can be created in this form? Or, have you talked to @lavalse about this and you both feel this approach to add multiple, but select the first one, makes sense?

File: web/src/components/molecules/Common/LinkAssetModal/LinkAssetModal.tsx [hunk 2]
```diff
@@ -144,6 +144,7 @@ const LinkAssetModal: React.FC<Props> = ({
           displayUploadModal={displayUploadModal}
           hideUploadModal={hideUploadModal}
           handleUpload={handleUpload}
+          afterClose={onCancel}
```
[the remark this answers is not in the document]
Comment (nourbalaha) [reply web/src/components/molecules/Common/LinkAssetModal/LinkAssetModal.tsx:147]: the method's name in ant design's website is `afterClose`, can you please check the api section on this [page](https://ant.design/components/modal).
if you think that `onClose` is better please let me know.

File: web/src/components/molecules/Common/Form/AssetItem/hooks.ts (hunk 1 above)
Comment (nourbalaha) [reply web/src/components/molecules/Common/Form/AssetItem/hooks.ts:36]: I am using the same fileUpload modal in asset list page and in schema and item. both use the same method `createAssets` which takes an array of files as an argument and returns an array of assets as a result.
However, in schema and item I already added `multiple: false` and `maxCount: 1` in the modal props, so it doesn't allows multiple file upload at.
please check [this](https://github.com/reearth/reearth-cms/blob/f1cc6f125ab70fd68a2e7b0f949461f910d928be/web/src/components/molecules/Common/Form/AssetItem/index.tsx#L59) for reference.

File: web/src/components/molecules/Content/Details/index.tsx [hunk 3]
```diff
@@ -21,7 +21,7 @@ export type Props = {
   onItemCreate: (data: { schemaId: string; fields: ItemField[] }) => Promise<void>;
   onItemUpdate: (data: { itemId: string; fields: ItemField[] }) => Promise<void>;
   onBack: (modelId?: string) => void;
-  createAssets: (files: UploadFile[]) => Promise<void>;
+  createAssets: (files: UploadFile[]) => Promise<(Asset | undefined)[]>;
```
[the remark this answers is not in the document]
Comment (KaWaite) [reply web/src/components/molecules/Content/Details/index.tsx:24]: OK

File: web/src/components/molecules/Common/LinkAssetModal/LinkAssetModal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (KaWaite) [reply web/src/components/molecules/Common/LinkAssetModal/LinkAssetModal.tsx:147]: We want our code to be Antd-centric ONLY where absolutely necessary, so only the antd component's props should be named according to antd conventions. Everything else should adhere to our standards

File: web/src/components/molecules/Common/Form/AssetItem/hooks.ts (hunk 1 above)
Comment (KaWaite) [reply web/src/components/molecules/Common/Form/AssetItem/hooks.ts:36]: Oh ok! Sounds good

File: web/src/components/molecules/Common/LinkAssetModal/LinkAssetModal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nourbalaha) [reply web/src/components/molecules/Common/LinkAssetModal/LinkAssetModal.tsx:147]: Ok, gotcha.

[the remark this answers is not in the document]
Comment (nourbalaha) [reply web/src/components/molecules/Common/LinkAssetModal/LinkAssetModal.tsx]: I rename `afterClose` to `onUploadModalClose`.

**Review by KaWaite — APPROVED** [review]

**Pull request merged**